File: Ocis.Client/sdk/py/Ocis_Server/protocol_spec.py
from dataclasses import dataclass
from ..fable_modules.fable_library.reflection import (enum_type, TypeInfo, uint32_type, uint8_type, int32_type, array_type, option_type, record_type, string_type)
from ..fable_modules.fable_library.types import (Record, uint8, uint32)

def _expr0() -> TypeInfo:
    return record_type("Ocis.Server.ProtocolSpec.RequestPacket", [], RequestPacket, lambda: [("MagicNumber", uint32_type), ("Version", uint8_type), ("CommandType", enum_type("Ocis.Server.ProtocolSpec.CommandType", int32_type, [("Set", 1.0), ("Get", 2.0), ("Delete", 3.0)])), ("TotalPacketLength", int32_type), ("KeyLength", int32_type), ("ValueLength", int32_type), ("Key", array_type(uint8_type)), ("Value", option_type(array_type(uint8_type)))])


@dataclass(eq = False, repr = False, slots = True)
class RequestPacket(Record):
    MagicNumber: uint32
    Version: uint8
    CommandType: enum_type("Ocis.Server.ProtocolSpec.CommandType", int, [("Set", 1.0), ("Get", 2.0), ("Delete", 3.0)])
    TotalPacketLength: int
    KeyLength: int
    ValueLength: int
    Key: bytearray
    Value: bytearray | None

RequestPacket_reflection = _expr0

def _expr1() -> TypeInfo:
    return record_type("Ocis.Server.ProtocolSpec.ResponsePacket", [], ResponsePacket, lambda: [("MagicNumber", uint32_type), ("Version", uint8_type), ("StatusCode", enum_type("Ocis.Server.ProtocolSpec.StatusCode", uint8_type, [("Success", 0.0), ("NotFound", 1.0), ("Error", 2.0)])), ("TotalPacketLength", int32_type), ("ValueLength", int32_type), ("ErrorMessageLength", int32_type), ("Value", option_type(array_type(uint8_type))), ("ErrorMessage", option_type(string_type))])


@dataclass(eq = False, repr = False, slots = True)
class ResponsePacket(Record):
    MagicNumber: uint32
    Version: uint8
    StatusCode: enum_type("Ocis.Server.ProtocolSpec.StatusCode", uint8, [("Success", 0.0), ("NotFound", 1.0), ("Error", 2.0)])
    TotalPacketLength: int
    ValueLength: int
    ErrorMessageLength: int
    Value: bytearray | None
    ErrorMessage: str | None

ResponsePacket_reflection = _expr1

__all__ = ["RequestPacket_reflection", "ResponsePacket_reflection"]

File: Ocis.Client/Ocis.Client.SDK/ProtocolSpec.fs
module Ocis.Client.SDK.ProtocolSpec

// NOTE: This file should be kept in sync with Ocis.Server.ProtocolSpec
// It is duplicated here to allow Fable compilation independently from Server

[<Literal>]
let MAGIC_NUMBER = 0x5349434Fu

[<Literal>]
let PROTOCOL_VERSION = 1uy

[<Literal>]
let HEADER_SIZE = 18

type CommandType =
  | Set = 1
  | Get = 2
  | Delete = 3

type StatusCode =
  | Success = 0uy
  | NotFound = 1uy
  | Error = 2uy

type RequestPacket =
  { MagicNumber : uint32
    Version : byte
    CommandType : CommandType
    TotalPacketLength : int32
    KeyLength : int32
    ValueLength : int32
    Key : byte array
    Value : byte array option }

type ResponsePacket =
  { MagicNumber : uint32
    Version : byte
    StatusCode : StatusCode
    TotalPacketLength : int32
    ValueLength : int32
    ErrorMessageLength : int32
    Value : byte array option
    ErrorMessage : string option }
